src/Calculator/Views/MainPage.xaml.cpp

```diff
@@ -601,3 +601,17 @@ void MainPage::AnnounceCategoryName()
     NarratorAnnouncement^ announcement = CalculatorAnnouncement::GetCategoryNameChangedAnnouncement(categoryName);
     NarratorNotifier->Announce(announcement);
 }
+
+void MainPage::OnNavItemInvoked(_In_ Microsoft::UI::Xaml::Controls::NavigationView^ sender, _In_ Microsoft::UI::Xaml::Controls::NavigationViewItemInvokedEventArgs^ e)
```

Comment:
Can you remove `Microsoft::UI::Xaml::Controls::`? 
(I would like Visual Studio to automatically check if `using namespace Microsoft::UI::Xaml::Controls` exists, but unfortunately, it's not the case :( )

Reply:
I was not able to remove `Microsoft::UI::Xaml::Controls`.

Reply:
Because the namespace is already declared , you can write:

`void MainPage::OnNavItemInvoked(_In_ NavigationView^ sender, _In_ NavigationViewItemInvokedEventArgs^ e)`

Reply:
I tried that. For some reason I got errors.

Reply:
@danbelcher-MSFT, I think you removed the `_In_` by mistake

`void MainPage::OnNavItemInvoked(_In_opt_  MUXC::NavigationView^ /*sender*/, _In_ MUXC::NavigationViewItemInvokedEventArgs^ e)`

Reply:
Actually, it was intentional.  `_In_opt_` describes a parameter that may or may not have a value for use by the function.  We are explicitly discarding this parameter so I figured there's no point to decorating it with SAL (imo).  `_In_opt_` doesn't seem to be the right fit and there isn't any other SAL equivalent.

Reply:
This is a (very) tricky question and open to debate.

Imo, I don't think it's a best practice to comment/discard a parameter in a function declaration (but it's totally fine with function definition), it makes the development harder when using IntelliSense (because the parameter is not named):

![image](https://user-images.githubusercontent.com/1226538/56257450-191eb000-6081-11e9-8639-2b20d47cf50f.png)

Another issue is virtual methods, if the method is overridden, the developer have no idea if the parameter is In, Out, In/Out, optional, etc...

```
class TestBase
{
public:
    virtual int TestFunc(int * /*test*/, _In_ int * test2) noexcept;
};

class Test: public TestBase
{
public:
    virtual int TestFunc(_InOut_ int * test, _In_ int * test2) noexcept;
};
```

in this case, a developer will have no idea if the first parameter is in/out/inout when using a TestBase* object.

For these 2 reasons (and probably more), I think we should not comment/discard a argument in a function declaration, so we need to use SAL for this parameter in the declaration.

Visual Studio forcing SAL to be in the function definition when set in the declaration, we are forced to do it in the function definition too.

So in our case:

.h
```
void OnNavItemInvoked(_In_ Microsoft::UI::Xaml::Controls::NavigationView^ sender...)
```

.cpp
```
void MainPage::OnNavItemInvoked(_In_ NavigationView^ /*sender.*/..)
```
 

Reply:
On the other hand, if we're consistent about naming parameters that are used and not naming parameters that are unused, then seeing a function with an unnamed parameter is a clear indication that the parameter is not used by that function.  In this case, I can simply supply `nullptr`.  Given that SAL describes how a parameter is used, for an unnamed parameter there would be no need for SAL.

This doesn't feel like an appropriate place for a long discussion of the semantics (I can see arguments for both sides).  In this case, let's let @cheezwhines resolve as they'd like and we revisit this elsewhere, if we have need.

Reply:
> then seeing a function with an unnamed parameter is a clear indication that the parameter is not used by that function.

This isn't true with virtual methods, a unused parameter in a base class can be in reality used by a function overriding it in a child class.

```
class WaterParkBase
{
public:
    virtual int GetPrice(_In_ int * numberAdults, int * /*numberKids*/) noexcept
    {
          m_price = *numberAdults * 50;
    }
...
};

class SFWaterPark: public c
{
public:
    virtual int GetPrice(_In_ int * numberAdults, _In_ int * numberKids) noexcept
    {
          m_price = *numberAdults * 35 + *numberKids *10;
    }
};
```
in this case:

```
SFWaterPark t = new SFWaterPark();
t->GetPrice(a, b);
``` 

Intellisense will show `GetPrice(int * numberAdults, int *)` even if `numberKids` will be used.

Reply:
Hey Rudy, I hear your perspective and I think I agree we should be consistent in naming parameters for declarations but remove them for definitions.  For now I'm going to take the change as is, to prevent more work for the author.  We'll endeavour to be consistent about this rule for future changes.  Thanks for sharing the examples!